Automation Examples/020.md

```diff
@@ -0,0 +1,161 @@
+selenium.open chrome url https://www.vat19.com/
+window ✱Unique✱ style maximize
+```
+
+Use window command and part of the window title as an argument to bring it to the front and maximise.
+
+#### 2. Go to fun & strange stuff, create lists which will contain product names and their prices
+
+After running the above script and seeing the webpage, you can inspect the button element and in the developer tools, right click it and choose "Copy Xpath" that you can insert in the search argument for selenium.click command.
+
+```G1ANT
+selenium.click search ‴//*[@id="side-nav-content"]/div[1]/ul/li[3]/a‴ by xpath
+list.create type text result ♥products
+list.create type text result ♥prices
+```
+
+#### 3. Wait for elements to be loaded, get link to the first image, get name of the product and price value and store them in the lists
+
+Right click the image element and see that the "src" attribute stores a link to it. The below JavaScript code returns a link to the first image. You can run it in the console to check it. 
+
+```js
+document.getElementsByClassName("lazyautosizes lazyloaded")[0].getAttribute("src")
+```
+
+So let's wait for this element using the below G1ANT code.
+
+```G1ANT
+selenium.waitforvalue ‴return document.getElementsByClassName("lazyautosizes lazyloaded").length > 0‴ expectedvalue true
+```
+
+After right clicking the product name and price, you may see that geting those values is as easy as below. 
+
+```G1ANT
+selenium.runscript ‴return document.getElementsByClassName("productSketch-name-link")[0].innerText‴ result ♥productName
+list.add ♥products toadd ♥productName
+selenium.runscript ‴return document.getElementsByClassName("regularPrice")[0].innerText‴ result ♥price
+list.add ♥prices toadd ♥price
+```
+
+#### 4. Donwload the first image
+
+In order to do this, add "https:" prefix to the "src" and use file.download command.
+
+```G1ANT
+♥imageSrc = ⊂"https:" + ♥imageSrc⊃
+file.download ♥imageSrc filename ♥basePath\1.png
+```
+
+#### 5. Put all above steps in a loop
+
+This is what you should get:
+
+```G1ANT
+for ♥i from 0 to 19
+    selenium.runscript ‴return document.getElementsByClassName("lazyautosizes lazyloaded")[♥i].getAttribute("src")‴ result ♥imageSrc
+    ♥imageSrc = ⊂"https:" + ♥imageSrc⊃
+    file.download ♥imageSrc filename ♥basePath\⊂♥i+1⊃.png
+    selenium.runscript ‴return document.getElementsByClassName("productSketch-name-link")[♥i].innerText‴ result ♥productName
+    list.add ♥products toadd ♥productName
+    selenium.runscript ‴return document.getElementsByClassName("regularPrice")[♥i].innerText‴ result ♥price
+    list.add ♥prices toadd ♥price
+end for
+```
+
+#### 6. After running this script, you may notice that it fails after some time
+
+It does so because webpage isn't fully loaded and some of the elements are not yet present so the script can't continue. To load more elements, you should scroll. Insert below line at the end of the loop to prevent this.
```

Comment:
"to prevent this"? 
i'd rather say "to achieve this"

Reply:
yeah hahahaha "mind shortcut" XD

Reply:
```suggestion
It does so because webpage isn't fully loaded and some of the elements are not yet present so the script can't continue. To load more elements, you should scroll. Insert below line at the end of the loop to achieve this.
```